From d14fc7d34d8d20ff2a6951b403e08d95d10f5eb2 Mon Sep 17 00:00:00 2001
From: Stephen L Arnold <sarnold@vctlabs.com>
Date: Sun, 8 Dec 2024 15:22:44 -0800
Subject: [PATCH] oe standard: cleanup passwd ageing and related login-defs
 checks

* add openembedded to common-passwd test, set pam passwd alg

Signed-off-by: Stephen L Arnold <sarnold@vctlabs.com>
---
 .../rule.yml                                  |  2 +-
 .../openembedded/profiles/standard.profile    | 21 +++++++++++++------
 shared/applicability/package.yml              | 11 +++++-----
 3 files changed, 22 insertions(+), 12 deletions(-)

diff --git a/linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/rule.yml b/linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/rule.yml
index 446d1abb3f..57e5c74223 100644
--- a/linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/rule.yml
+++ b/linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/rule.yml
@@ -2,7 +2,7 @@ documentation_complete: true
 
 title: "Set PAM''s Password Hashing Algorithm"
 
-{{% if product in ["sle12", "sle15"] or 'ubuntu' in product %}}
+{{% if product in ["sle12", "sle15", "openembedded"] or 'ubuntu' in product %}}
 {{% set pam_passwd_file_path = "/etc/pam.d/common-password" %}}
 {{% else %}}
 {{% set pam_passwd_file_path = "/etc/pam.d/system-auth" %}}
diff --git a/products/openembedded/profiles/standard.profile b/products/openembedded/profiles/standard.profile
index 37547a454e..f889faf673 100644
--- a/products/openembedded/profiles/standard.profile
+++ b/products/openembedded/profiles/standard.profile
@@ -66,22 +66,29 @@ selections:
     - accounts_password_pam_minlen
     - accounts_password_pam_retry
     - var_password_pam_minclass=4
-    - var_password_pam_minlen=14
+    - var_password_pam_minlen=10
+    - var_password_pam_retry=3
     - accounts_password_pam_pwhistory_remember_password_auth
     - accounts_password_pam_pwhistory_remember_system_auth
-    - var_password_pam_remember_control_flag=required
-    - var_password_pam_remember=5
+    - accounts_password_pam_unix_remember
+    - var_password_pam_remember=24
+    - var_password_hashing_algorithm=SHA512
+    - var_password_hashing_algorithm_pam=sha512
     - set_password_hashing_algorithm_systemauth
     - var_accounts_maximum_age_login_defs=365
+    # Needs variable: var_accounts_password_set_max_life_existing=365
+    - accounts_maximum_age_login_defs
     - accounts_password_set_max_life_existing
-    - var_accounts_minimum_age_login_defs=7
+    - var_accounts_minimum_age_login_defs=1
+    # Needs variable: var_accounts_password_set_min_life_existing=1
+    - accounts_minimum_age_login_defs
     - accounts_password_set_min_life_existing
     - var_accounts_password_warn_age_login_defs=7
+    - accounts_password_warn_age_login_defs
     - account_disable_post_pw_expiration
     - var_account_disable_post_pw_expiration=30
     - no_shelllogin_for_systemaccounts
-    - accounts_tmout
-    - var_accounts_tmout=15_min
     - accounts_root_gid_zero
     - accounts_umask_etc_profile
     - use_pam_wheel_for_su
@@ -109,6 +116,8 @@ selections:
     - motd_banner_text=cis_banners
     - banner_etc_issue
     - login_banner_text=cis_banners
+    - banner_etc_issue_net
+    - remote_login_banner_text=cis_banners
     - file_groupowner_etc_motd
     - file_owner_etc_motd
     - file_permissions_etc_motd
diff --git a/shared/applicability/package.yml b/shared/applicability/package.yml
index 2cade2416a..2d28325181 100644
--- a/shared/applicability/package.yml
+++ b/shared/applicability/package.yml
@@ -75,12 +75,13 @@ args:
     pkgname: openssh
   pam:
     {{% if pkg_system == "rpm" %}}
-      pkgname: pam
-      {{% if product in ["openembedded"] %}}
-      pkgname: libpam-runtime
-      {{% endif %}}
+    {{% if product in ["openembedded"] %}}
+    pkgname: libpam-runtime
+    {{% else %}}
+    pkgname: pam
+    {{% endif %}}
     {{% else %}}
-      pkgname: libpam-runtime
+    pkgname: libpam-runtime
     {{% endif %}}
   pam_apparmor:
     pkgname: pam_apparmor
-- 
2.47.0

